src/List/ListItem.js

```diff
@@ -41,6 +42,7 @@ function getStyles(props, context, state) {
       !state.rightIconButtonKeyboardFocused ? hoverColor : null,
       color: textColor,
       display: 'block',
+      cursor: disabled ? 'not-allowed' : 'inherit',
```

Comment:
And I think we should apply cursor changing here.

As you can see, calculation the cursor style is based on component props.

I agree that  it will be better also change the styles to make disabled state more prominent and obvious, but:
- I want to make pull-request as small as possible, and changing style can be a new issue with own pull-request7
- changing the cursor is also important to indicate component disabled state and disabled interaction
- [original issue](https://github.com/callemall/material-ui/issues/3817)  was created to keep consistency between components. If any component will not have cursor change in disabled state - it is not consistency, as for me.